### PR TITLE
[NEW] Improve DayOfEvent Components UI

### PR DESCRIPTION
**Pull request by Dnouv**
Hello,

This PR works on introducing a new UI for the Greenroom page (previously at `virtualconf/greenroom`) now at (`conferences/greenroom/1`).

Changes are done (to be updated as we progress):

- [x] Introduce a common background wrapper component to be used for both the mainstage and greenroom
- [x] Remove unnecessary and confusing buttons
- [x] Remove the current non-working chat component
- [x] Placed a placeholder for the chat component
- [x] Remove unnecessary CSS files
- [x] Remove unnecessary Jitsi buttons
- [x] Add required functionalities using Jitsi API (mic mute, camera mute, and device settings)
- [x] Revamp Mainstage page UI
- [x] Integrate Chat Component on both the Mainstage and Greenroom page
- [x] Add the People tab (showing current joined speakers) on the Speaker Greenroom Page
- [x] Add validation on mainstage page for verifying logged in user
- [x] Improve the dummy auth to login as "Admin, Speaker, and Attendee" roles
- [x] Add support for RC Auth.
- [x] Make Pages responsive to smaller screens
- [x] Refactor build errors using middlewares
- [x] Refactor Dummy Auth

Current roadblocks:

- [x] External call to `toggleScreenShare` in jitsi is not working.

Next to be implemented in another PR:

- [ ] Add Streaming switches

Some preview images,

image

Screen Capture:

greenroom_demo

Fixes: #12 

Thank you!

### CODE REVIEW

**Review by Sing-Li — CHANGES_REQUESTED** [review]
Please add @sidmohanty11  's   actual embedded chat immediately.

We should not be working with placeholders at this stage of the project.

make sure auth + superprofile + permission to access  are all working across the components

and that membership of the viewers  (plus the speakers and admin) are visible in the greenroom

**Comment by Dnouv** [comment]
The greenroom Page,

https://user-images.githubusercontent.com/61188295/184314786-1ddc40ae-631d-457f-9703-ce2f17abdf40.mp4

The Mainstage Page and the GreenRoom page chat sync demo,


https://user-images.githubusercontent.com/61188295/184314866-0b3cd3c9-fe6e-4a79-adaf-a45ba0283b70.mp4

Feedbacks are most welcomed. Thank you!

**Comment by Dnouv** [comment]
Why did I do it this way?

- Providing a waiting lobby/room for the speaker to test out devices and also adjust their name/ virtual background is most needed. Since joining in a live stream and then changing the camera/microphone would be a negative UX for both the Attendees and the speaker themselves. I strongly agree providing many buttons there would confuse the users more, and as such, all the weird buttons are removed from the prejoin page.
- Left the End Call Button since it was pretty simple and won't break or confuse the Speakers in any way; please correct me if I misunderstood this.
- I still think providing the users with the device names is a good addition; they just need to select one among them which is working, and it rolls back to the previous device in case of any errors, which in a way notifies the user there is a problem with this specific selection. And in this way, we are making it hard for the users with at least a basic level of tech understanding since they'd prefer to select the device by name.
- In the case of only the "Next Device", the user won't have any idea if the changed device is working or not (yeah, I looked for a way to provide audio level graphs, but it'd take around a week to add.)

Thank you!

**Comment by Sing-Li** [comment]
@Dnouv  please provide some information on how to test this.    

I logged in using the dummy auth and went to event menus - no matter what I do, I cannot reach the greenroom OR the mainstage with video + chat.

Please add the code to make the conditional showing of these menu test-able.    (perhaps seeding a user with admin role in the database ?)

**Comment by Sing-Li** [comment]
Because you did data-seeding on another PR, @Dnouv  - I've quick merged that one so you can fix this.

We need to  test with THREE different pre-seeded credentials --- one with role of  `speaker,` one as `admin`, one as non-speaker, non-admin     They should all meet with the role-appropriate  menu options as we agreed a few weeks ago.

You may have to trivially modify the _dummy auth_   to support `login as admin` ,  `login as speaker`,  and `login`    FWIW

**Comment by Sing-Li** [comment]
Please note that I will also test with a FOURTH CONIDTION.   That of not logged in!

In this case,  perhaps someone shared Duda's promoted demo day URL --- the user, not logged on at all, should still see without restriction:

   - the poster with the agenda
   
But   will not be able to enter or see the mainstage until creating an account and logging in.

**Comment by Sing-Li** [comment]
@Dnouv   this branch currently does not work.

The `Login as` UI just freezes after you click one of the roles.   And many errors on the server side follows.

image



```
wait  - compiling /api/auth/[...nextauth]...
event - compiled successfully in 50 ms (139 modules)
[next-auth][warn][NEXTAUTH_URL]
https://next-auth.js.org/warnings#nextauth_url
[next-auth][warn][NO_SECRET]
https://next-auth.js.org/warnings#no_secret
wait  - compiling /virtualconf/mainstage (client and server)...
event - compiled client and server successfully in 323 ms (978 modules)
[2022-08-14 16:52:24.524] http: GET /api/speakers (16 ms) 200
RC4Community is now using a dummy Auth Component! If you wish to use a robust Auth component, provide all the credentials first (https://github.com/RocketChat/RC4Community/tree/master/app/components/auth)
animation rerendered
Warning: Each child in a list should have a unique "key" prop.

Check the top-level render call using <span>. See https://reactjs.org/link/warning-keys for more information.
    at p
    at div
    at /home/autoqa/Dev/RC4Conferences/app/node_modules/react-bootstrap/cjs/Col.js:72:8
    at div
    at /home/autoqa/Dev/RC4Conferences/app/node_modules/react-bootstrap/cjs/Row.js:21:3
    at div
    at /home/autoqa/Dev/RC4Conferences/app/node_modules/react-bootstrap/cjs/Container.js:24:3
    at main
    at Mainstage (webpack-internal:///./pages/virtualconf/mainstage/index.js:36:22)
    at Layout (webpack-internal:///./components/layout.js:22:81)
    at SessionProvider (/home/autoqa/Dev/RC4Conferences/app/node_modules/next-auth/react/index.js:416:24)
    at ApolloProvider (/home/autoqa/Dev/RC4Conferences/app/node_modules/@apollo/client/react/context/context.cjs:50:21)
    at $29383e587d62412a$export$9f8ac96af4b1b2ae (/home/autoqa/Dev/RC4Conferences/app/node_modules/@react-aria/ssr/dist/main.js:25:28)
    at MyApp (webpack-internal:///./pages/_app.js:30:18)
    at StyleRegistry (/home/autoqa/Dev/RC4Conferences/app/node_modules/next/dist/styled-jsx/dist/index/index.js:671:34)
    at AppContainer (/home/autoqa/Dev/RC4Conferences/app/node_modules/next/dist/server/render.js:280:29)
    at AppContainerWithIsomorphicFiberStructure (/home/autoqa/Dev/RC4Conferences/app/node_modules/next/dist/server/render.js:309:57)
    at div
    at Body (/home/autoqa/Dev/RC4Conferences/app/node_modules/next/dist/server/render.js:596:21)
wait  - compiling /api/auth/[...nextauth]...
event - compiled successfully in 19 ms (131 modules)
upstream image response failed for https://media-exp1.licdn.com/dms/image/C5103AQEfuswPxV6Gyw/profile-displayphoto-shrink_800_800/0/1559890882906?e=1654128000&v=beta&t=Vvuss_1J3PNLb0T-M9kCQ2emP_4KonboB16Yrnu1gB4 403
upstream image response failed for https://media-exp1.licdn.com/dms/image/C5603AQGio-wOX6rWyA/profile-displayphoto-shrink_800_800/0/1599022697338?e=1654128000&v=beta&t=4eevWzts6GpDF_FELQuP1bD9HDx17cO1ChDL3fjOPl4 403
upstream image response failed for https://media-exp1.licdn.com/dms/image/C5103AQF7LAGv0_LeBg/profile-displayphoto-shrink_800_800/0/1548759328186?e=1654732800&v=beta&t=P-k5uxD1J955b7eXjwE20_IGlABM0tR0158cdcVaSFE 403
[2022-08-14 16:53:10.404] http: GET /api/speakers (14 ms) 200
RC4Community is now using a dummy Auth Component! If you wish to use a robust Auth component, provide all the credentials first (https://github.com/RocketChat/RC4Community/tree/master/app/components/auth)
animation rerendered
upstream image response failed for https://media-exp1.licdn.com/dms/image/C5103AQEfuswPxV6Gyw/profile-displayphoto-shrink_800_800/0/1559890882906?e=1654128000&v=beta&t=Vvuss_1J3PNLb0T-M9kCQ2emP_4KonboB16Yrnu1gB4 403
wait  - compiling / (client and server)...
event - compiled client and server successfully in 147 ms (1009 modules)
[2022-08-14 16:53:21.061] http: GET /api/carousels (6 ms) 200
[2022-08-14 16:53:21.067] http: GET /api/personas (3 ms) 200
[2022-08-14 16:53:21.072] http: GET /api/guide (3 ms) 200
[2022-08-14 16:53:21.077] http: GET /api/release-note (3 ms) 200
[2022-08-14 16:53:21.087] http: GET /api/top-nav-item (8 ms) 200
RC4Community is now using a dummy Auth Component! If you wish to use a robust Auth component, provide all the credentials first (https://github.com/RocketChat/RC4Community/tree/master/app/components/auth)
[2022-08-14 16:54:15.935] http: GET /api/speakers (10 ms) 200
RC4Community is now using a dummy Auth Component! If you wish to use a robust Auth component, provide all the credentials first (https://github.com/RocketChat/RC4Community/tree/master/app/components/auth)
animation rerendered
upstream image response failed for https://media-exp1.licdn.com/dms/image/C5103AQEfuswPxV6Gyw/profile-displayphoto-shrink_800_800/0/1559890882906?e=1654128000&v=beta&t=Vvuss_1J3PNLb0T-M9kCQ2emP_4KonboB16Yrnu1gB4 403
```

File: cleanup.sh [hunk 1]
```diff
@@ -5,14 +5,14 @@ echo "--Stopping the Open Event server--"
 
 if [ ! -z $OES_CONTAINER_ID ]; then
     cd open-event-server
-    docker compose down -v
```
Comment (Sing-Li) [inline cleanup.sh:8]: As we have discussed.   You *cannot* assume the dev / contributor have one or the other.   

Make sure the script tests and work for both!

**Review by Sing-Li — CHANGES_REQUESTED** [review]

**Comment by Sing-Li** [comment]
And please test with login from the four different roles.   (do not test with direct URL)   It is vital to the usability of this system.  Thanks.

**Pull request merged**